### PR TITLE
Add .mvn/nisse.properties with export-subst for source-archive builds

### PR DESCRIPTION
**Pull request by gnodet**
## Summary

- Adds `.gitattributes` marking `.mvn/nisse.properties` for `export-subst`
- Adds `.mvn/nisse.properties` with `nisse.jgit.dynamicVersion` and `nisse.jgit.date` as low-priority fallback values

This enables `git archive` source tarballs to build without a `.git` directory. The `$Format:...$` placeholders are expanded by `git archive` into the described tag and commit date, which nisse 0.9.5+ (this project uses 0.9.7) reads as a lowest-priority property source.

Normal git checkout builds are unaffected — JGit resolves the version directly and these fallback values are ignored.

See [maveniverse/nisse#182](https://github.com/maveniverse/nisse/issues/182) and [jline/jline3#2153](https://github.com/jline/jline3/issues/2153) for context.

## Test plan

- [x] `./mvnw clean install -B -DskipTests` — git checkout build: **BUILD SUCCESS** (version `0.3.11-32-SNAPSHOT`)
- [x] `git archive --prefix=test/ HEAD | tar -xf - -C /tmp/ && cd /tmp/test && ./mvnw clean install -B -DskipTests` — source archive build (no `.git`): **BUILD SUCCESS** (version `0.3.10-12-gb314e1e` from expanded nisse.properties)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Maintenance**
  * Improved version and commit date information when building from archived source releases.
  * Added fallback values to keep build metadata available in exported archives.
  * Standardized commit dates using ISO 8601 format for more consistent build information.

* **Tests**
  * Added automated checks to verify consistent version resolution between Git checkouts and source archives.
  * Expanded validation for tagged and untagged source archives, including builds without reachable tags.

### CODE REVIEW

File: src/test/scripts/verify-nisse-archive-parity.sh [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    printf '  (expected: formats differ between JGit dynamic version and git-describe)\n'
+fi
+
+printf '\n=== Test 4: no-reachable-tag guard ===\n'
+# Verify that %(describe:tags=true) expands to empty when no tag is reachable,
+# and document this as a known limitation.
+ORPHAN_DIR="$WORK_DIR/orphan-repo"
+mkdir -p "$ORPHAN_DIR"
+git -C "$ORPHAN_DIR" init -q
+git -C "$ORPHAN_DIR" commit --allow-empty -m "orphan" -q
```
Comment (gnodet) [inline src/test/scripts/verify-nisse-archive-parity.sh]: **Low:** The orphan commit relies on a globally configured `user.name` and `user.email`. In clean CI or container environments with no global git identity, this will fail with exit code 128.

```suggestion
  git -C "$ORPHAN_DIR" init -q
  git -C "$ORPHAN_DIR" -c user.name=test -c user.email=test@test commit --allow-empty -m "orphan" -q
```

**Review by gnodet — COMMENTED** [review]
Well-crafted build infrastructure PR that correctly uses nisse's native `.mvn/nisse.properties` fallback mechanism to enable source-archive builds. Excellent in-file documentation, sound design, and all CI checks pass.

This is the second attempt at enabling source-archive builds; the new approach correctly uses `.mvn/nisse.properties` (the nisse-native fallback file supported since nisse 0.9.5) which avoids the property-priority conflicts that caused the first attempt to be reverted.

**Minor observations (non-blocking):**

1. **Test script git identity** — The orphan commit in the verify script (`git commit --allow-empty`) relies on a globally configured `user.name`/`user.email`. In clean CI or container environments, this will fail. Consider adding `-c user.name=test -c user.email=test@test`.

2. **Swallowed stderr in `resolve_version()`** — stderr is redirected to `/dev/null`, which silently swallows Maven errors. If Maven fails for unrelated reasons, the script would report an empty version and misattribute the failure.

3. **Test script not in CI** — The script serves as a manual developer verification tool, which is fine, but worth noting for maintainers.

### 📋 PR Metadata

| Aspect | Current | Suggested |
|--------|---------|----------|
| Category | _(unlabeled)_ | `chore` |
| Milestone | _(none)_ | `0.3.11` |

_This review was generated by an AI agent and may contain inaccuracies. Please verify all suggestions before applying._

_Claude Code on behalf of Guillaume Nodet_

**Review by gnodet — COMMENTED** [review]
Well-crafted build infrastructure PR that uses nisse's native `.mvn/nisse.properties` fallback to enable source-archive builds. The approach correctly avoids the property-priority conflict that caused the prior attempt (e15f63d) to be reverted, with thorough inline documentation and a comprehensive verification script.

### Findings (informational, low severity)

1. **Git identity in test script** — `verify-nisse-archive-parity.sh` line ~82: The orphan commit (`git commit --allow-empty`) requires globally configured `user.name` and `user.email`. In clean CI containers or ephemeral environments without a git identity, this test will fail. Adding `-c user.name=test -c user.email=test@test` would make it portable:
   ```bash
   git -c user.name=test -c user.email=test@test -C "$ORPHAN_DIR" init -q
   git -c user.name=test -c user.email=test@test -C "$ORPHAN_DIR" commit --allow-empty -m "orphan" -q
   ```

2. **Stderr redirect in resolve_version()** — `verify-nisse-archive-parity.sh` line ~22: `resolve_version()` redirects stderr to `/dev/null`, silently swallowing Maven errors. If Maven fails for an unrelated reason, the function returns an empty string and the script misattributes the failure rather than surfacing the actual error.

### General Comments

- The approach of using `.mvn/nisse.properties` (lowest-priority property source in nisse 0.9.5+) instead of the previously reverted `.mvn/maven-user.properties` (high priority) is the right fix.
- The inline documentation in `nisse.properties` is excellent — it explains both the mechanism and the known limitations.
- The test script covers the important scenarios: checkout build, archive build, tagged-commit parity, and no-reachable-tag edge case.
- Clean commit history showing iterative refinement based on feedback.

### 📋 PR Metadata

| Aspect | Current | Suggested |
|--------|---------|----------|
| Category | _(unlabeled)_ | `chore` |
| Milestone | _(none)_ | `0.3.11` |

---

_This review was generated by an AI agent and may contain inaccuracies. Please verify all suggestions before applying._

_Claude Code on behalf of Guillaume Nodet_
